umbral-pre-python/umbral_pre/__init__.pyi

```diff
@@ -123,14 +126,15 @@ def decrypt_original(delegating_sk: SecretKey, capsule: Capsule, ciphertext: byt
     ...
 
 
+@final
 class KeyFrag:
 
     def verify(
             self,
             verifying_pk: PublicKey,
-            delegating_pk: Optional[PublicKey],
-            receiving_pk: Optional[PublicKey],
-            ) -> VerifiedKeyFrag:
+            delegating_pk: Optional[PublicKey] = None,
+            receiving_pk: Optional[PublicKey] = None,
```

Comment:
👍 - that looks better.

Reply:
I don't think it is correct though, Rust extension methods don't have defaults for parameters.

Reply:
Not for python bindings? Ok, so then this is just supposed to be `receiving_pk: Optional[PublicKey]`?

Reply:
Actually wait, I was wrong. Evidently, `Option` args are an exception, PyO3 adds default values to them: https://pyo3.rs/main/function/signature#function-signatures